### PR TITLE
Support dynamic modification of template

### PR DESCRIPTION
**Pull request by wanjunlei**
Signed-off-by: wanjunlei <wanjunlei@kubesphere.io>

### Support the dynamic modification of the template.
Now notification manager can dynamically update the template by directly reading the configmap where the template file is located, instead of mounting the configmap to the notification manager pod.

```
apiVersion: notification.kubesphere.io/v2beta1
kind: NotificationManager
metadata:
  name: notification-manager
spec:
  template:
    text:
      name: notification-manager-template
      namespace: kubesphere-monitoring-system
```

### Receivers can add template files for themselves.

The receiver can add their own templates like this.

```
apiVersion: notification.kubesphere.io/v2beta2
kind: Receiver
metadata:
  labels:
    type: tenant
    user: admin
  name: admin-email-receiver
spec:
  email:
    enabled: true
    tmplText:
      name: template-admin
      namespace: default
      key: template
    to:
    - admin@kubesphere.io
```

> `name` is the name of the configmap where the template file is located. If `key` is not set, all files in the configmap will be treated as template files.

> If the receiver defines a template with the same name as the global template, the global template will be overwritten. The overwrite operation will only affect the current receiver.

### Support for custom notification message languages.

Now it can custom notification message languages like this.

1. Define a language pack like this.

```
apiVersion: v1
data:
  zh-cn: |
    - name: zh-cn
      dictionary:
        alert: "告警"
        alerts: "告警" 
        firing: "触发中"
        resolved: "已解决"
        alertname: "告警名称"
        alerttype: "告警类型"
        cluster: "集群"
        namespace: "项目"
        severity: "告警级别"
        container: "容器"
        pod: "容器组"
        service: "服务"
        deployment: "部署"
        job: "任务"
        daemonset: "守护进程集"
        statefulset: "有状态副本集"
        instance: "实例"
        resource: "资源"
        user: "用户"
        verb: "操作"
        group: "用户组"
        requestReceivedTimestamp: "请求接收时间"
        role: "角色"
        host_ip: "主机IP"
        node: "节点"
        rule_id: "告警规则"
        owner_kind: "目标类型"
        workload: "工作负载"
kind: ConfigMap
metadata:
  name: zh-cn
  namespace: kubesphere-monitoring-system
```

2. Modify the notification manager cr.

```
apiVersion: notification.kubesphere.io/v2beta1
kind: NotificationManager
metadata:
  name: notification-manager
spec:
  template:
    language: zh-cn
    languagePack:
    - name: zh-cn
       namespace: kubesphere-monitoring-system
```

The notification will be like this.

### CODE REVIEW

**Comment by benjaminhuo** [comment]
Why the global's is called `template` while the receiver's is called`tmplText`
What's the difference of language and languagepack?

**Comment by wanjunlei** [comment]
> Why the global's template is called `template` while the receiver's is called`tmplText` What's the difference between language and languagepack?

The whole global's template is this.

```
template:
    text:
      name: notification-manager-template
      namespace: kubesphere-monitoring-system
   language: zh-cn
   languagePack:
   -  name: zh-cn
      namespace: kubesphere-monitoring-system
```

Actually, the global template is `template.text`, In order not to make breaking changes to the receiver crd, receiver's template is called `tmplText`.

`languagePack` is list of configmap which the language dictionary be in, `language` define which language to use.

**Comment by benjaminhuo** [comment]
You mean the receiver's tmplText could be the same as global's, but to avoid break old version, this is set to tmplText?

**Comment by wanjunlei** [comment]
> You mean the receiver's tmplText could be the same as global's, but to avoid break old version, this is set to tmplText?

Yes.

File: pkg/constants/constants.go [hunk 1]
```diff
@@ -16,4 +16,19 @@ const (
...
+
+	AlertFiring   = "firing"
+	AlertResolved = "resolved"
+
+	AlertName      = "alertname"
+	AlertType      = "alertype"
+	AlertTime      = "alertime"
+	AlertMessage   = "message"
+	AlertSummary   = "summary"
+	AlertSummaryCN = "summaryCn"
```
Comment (benjaminhuo) [inline pkg/constants/constants.go:28]: summaryCN

File: pkg/internal/common.go [hunk 2]
```diff
@@ -13,6 +13,13 @@ const (
 	priority = "priority"
 )
 
+type Template struct {
+	TmplName      string                        `json:"tmplName,omitempty"`
+	TitleTmplName string                        `json:"titleTmplName,omitempty"`
```
Comment (benjaminhuo) [inline pkg/internal/common.go:18]: TitleTmplName => SubjectTmplName

File: pkg/constants/constants.go (hunk 1 above)
Comment (wanjunlei) [reply pkg/constants/constants.go:28]: The event rule use `summaryCn`.

https://github.com/kubesphere/kube-events/blob/master/helm/templates/crs/cluster-rules-default.yaml#L33

**Review by benjaminhuo — APPROVED** [review]

**Pull request merged**